src/flags.rs

```diff
@@ -94,21 +98,24 @@ impl Flags {
             None => usize::max_value(),
         };
 
-        let blocks: Vec<Block> = if !blocks_inputs.is_empty() {
```

Comment:
Any reason why you switched this from if to match?

Reply:
I was trying to add `inode` to this `if else` chain at first, but found that a `switch case` may be more suitable.

after googling, `match` like this is how `switch` do in `rust`.

Or could you leave some advice? Rust is all new to me

Reply:
For `inode` I think the current way in which you have done would be the best option. As you said, it would be too many ifs or nested matches, both not ideal.

I just think that if is a more readable than match here. This match is more like "match all but if (condition)" which is essentially if. Could you change this back and. Once that is done, I think we are good to go.